components/Onboarding.js

```diff
@@ -0,0 +1,217 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
+import I18n from 'ex-react-native-i18n';
+import Icon from './Icon';
+import colours from '../styles/colours';
+import fonts from '../styles/fonts';
+import { step8Top, step9Top } from '../styles/onboardingHelper';
+import { ACTION_STEPS, BUTTON_STEPS, STEPS, ORDERED_STEPS } from '../assets/constants/OnboardingSteps';
+
+const containerStyles = StyleSheet.create({
+  defaultContainer: {
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  step6Container: {
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+    paddingBottom: 20,
+  },
+  step7Container: {
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+    paddingBottom: 20,
+  },
+  step9Container: {
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+    paddingBottom: 20,
+  },
+});
+
+const highlightButtonStyles = StyleSheet.create({
+  step8Button: {
+    position: 'absolute',
+    top: step8Top(),
+    left: 3,
+    width: 35,
+    height: 35,
+    borderWidth: 3,
+    borderColor: colours.blues.base,
+    borderRadius: 5,
+  },
+  step9Button: {
+    position: 'absolute',
+    top: step9Top(),
+    left: 15,
+    width: 210,
+    height: 35,
+    borderWidth: 3,
+    borderColor: colours.blues.base,
+    borderRadius: 5,
+  },
+});
+
+const styles = StyleSheet.create({
+  container: {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    height: '100%',
+    width: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'center',
+    zIndex: 999,
+  },
+  onBoardingContainer: {
+    width: '80%',
+    paddingVertical: 10,
+    paddingHorizontal: 15,
+    borderRadius: 20,
+    borderWidth: 1,
+    borderColor: colours.greys.lighter,
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    backgroundColor: colours.white,
+  },
+  quitBtn: {
+    position: 'absolute',
+    top: 5,
+    right: 10,
+  },
+  onBoardingInfo: {
+    flexDirection: 'column',
+    alignItems: 'center',
+    paddingBottom: 10,
+  },
+  infoHeader: {
+    fontSize: fonts.header,
+    paddingBottom: 5,
+    textDecorationLine: 'underline',
+  },
+  infoDescription: {
+    fontSize: fonts.large,
+  },
+  buttonContainer: {
+    width: '100%',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  actionBtn: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  actionBtnText: {
+    fontSize: fonts.large,
+    color: colours.blues.base,
+  },
+});
+
+class Onboarding extends PureComponent {
+  componentDidUpdate() {
+    this.props.stepAction();
```

Comment:
⚠️ No conditions? You wild man! Was this intentional?

Reply:
Haha well, the conditions are checked within the actual function, but I'll take another look and make sure that its contained well